c/ztest.cc

```diff
@@ -1,18 +1,37 @@
 //------------------------------------------------------------------------------
-// This Source Code Form is subject to the terms of the Mozilla Public
-// License, v. 2.0. If a copy of the MPL was not distributed with this
-// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+// Copyright 2019 H2O.ai
 //
-// © H2O.ai 2018
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 //------------------------------------------------------------------------------
 #ifdef DTTEST
 #include "datatablemodule.h"
-#include "python/args.h"
 #include "utils/exceptions.h"
 #include "ztest.h"
 namespace py {
 
 
+/**
+ * Helper function to check for missing positional arguments
+ */
+static void check_posonly_args(const PKArgs& args_spec, const PKArgs& args) {
```

Comment:
This could be a method on `PKArgs`. In fact, there could even be a property such as `num_required_args`, which would check those arguments automatically, though I'm afraid it'll make the constructor even more complicated

Reply:
Are there any cases when number of arguments received is less than `n_posonly_args`?

Reply:
technically any arguments (including positional) can be optional

Reply:
Btw, what about also adding a check to `PKArgs::operator[]`?
```
const Arg& PKArgs::operator[](size_t i) const {
  //check here
  return bound_args[i];
}
```
It seems that this is a source of many segfaults we've seen in the past and also the issue with these cpp tests.

Reply:
Can add an `xassert()` there